varinfo/var_info.py

```diff
@@ -138,17 +130,7 @@ def _set_mission_and_short_name(self):
 
         """
         if self.short_name is None:
-            self.short_name = next(
-                (
-                    recursive_get(self.global_attributes, split_attribute_path(item))
-                    for item in self.var_info_config.get(
-                        'Collection_ShortName_Path', []
-                    )
-                    if recursive_get(self.global_attributes, split_attribute_path(item))
-                    is not None
-                ),
-                None,
-            )
+            self._set_short_name()
```

Comment:
Okay, so this is a change that I hadn't quite counted on going in...

Prior to this change, the global metadata attributes for the input were being parsed. Then, if the short name was not supplied as an argument to `VarInfoFrom*`, those global metadata attributes were checked for places we know data providers like to put the short name.

Now, we don't have those global attributes, so what happens now:

* If the short name is not already given...
* The source file is opened and the locations we think the short name might be are checked.

It's a little clunky where the DMR or NetCDF-4 file is now opened twice, but I couldn't think of a clever way around that.

Reply:
I'm not sure I understand this comment. Do you mean the only place a string specifying where a short name metadata attribute might be in the configuration file? If so, yes. (But the preference is still for an end-user to use the `short_name` kwarg when instantiating a `VarInfoFromDmr` or `VarInfoFromNetCDF4` object)

With regards to the tests, I'm not sure but maybe `tests.unit.test_var_info::test_var_info_short_name` might cover some of what you're looking for?

Reply:
this is the last thing I'm looking at, this is probably a me problem.

Reply:
🔔 And I got it.